vm_phys.c

```diff
@@ -0,0 +1,251 @@
+#include <vm_phys.h>
+#include <queue.h>
+#include <kmem.h>
+#include <libkern.h>
+
+
+#define ORDER_SIZE(x) (1 << (x))
+
+/* this is a stub, should be moved to other file */
+#define panic(MSG...) do { kprintf(MSG); while(1); } while (0);
+
+TAILQ_HEAD(vm_seglist, vm_phys_seg);
+static struct vm_seglist seglist;
+
+void vm_phys_print_free_blocks() {
+    struct vm_phys_seg *seg_it;
+    TAILQ_FOREACH(seg_it, &seglist, segq)
+    {
+        int i;
+        for(i = 0; i < VM_NFREEORDER; i++) {
+            kprintf("order %d:\n", i);
+            struct vm_page *pg_it;
+            TAILQ_FOREACH(pg_it, &seg_it->free_queues[i], freeq)
+                kprintf("(%p,%p) ", pg_it->phys_addr, pg_it->phys_addr+ORDER_SIZE(pg_it->order) );
+            kprintf("\n");
+        }
+    }
+    kprintf("\n");
+}
+
+void vm_phys_add_seg(vm_paddr_t start, vm_paddr_t end)
+{
+    struct vm_phys_seg *seg = km_early_alloc(sizeof(struct vm_phys_seg));
+    seg->start = start;
+    seg->end = end;
+    int page_array_size = (seg->end - seg->start) / PAGE_SIZE;
+    seg->page_array = km_early_alloc(page_array_size);
+    TAILQ_INSERT_TAIL(&seglist, seg, segq);
+    if(ALIGN(seg->start, PAGE_SIZE) != (uintptr_t) seg->start)
+        panic("vm_phys_add_seg: start not page aligned ")
+    if(ALIGN(seg->end, PAGE_SIZE) != (uintptr_t) seg->end)
+        panic("vm_phys_add_seg: end not page aligned ")
+
+    int i;
+    for(i = 0; i < page_array_size; i++) {
```

Comment:
If I don't merge e07f0d097eed568e6b3d06a46eead99a51ab12f4 then it does complain. Please rebase your branch.